core/src/main/java/net/opentsdb/query/pojo/TimeSeriesQuery.java

```diff
@@ -762,7 +762,7 @@ private ExecutionGraphNode downsampler(final Metric metric,
     DownsampleConfig.Builder ds = (DownsampleConfig.Builder) DownsampleConfig.newBuilder()
         .setAggregator(downsampler.getAggregator())
         .setInterval(downsampler.getInterval())
-        .setFill(downsampler.getFillPolicy() != null ? true : false)
+        .setFill(policy != null && policy != FillPolicy.NONE ? true : false)
```

Comment:
`policy != null && policy != FillPolicy.NONE` is enough. No need of `?true:false`